### PR TITLE
Update to Compat_OPTRE

### PR DESCRIPTION
**Pull request by Nathan423**
added stingtable.xml and more selections for the biofoam. 
Adjusted the weight of missiles + ammo for new stamina system.

### CODE REVIEW

File: optionals/compat_optre/ACE_Medical_Actions.hpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
+class ACE_Head {
```
Comment (jonpas) [inline optionals/compat_optre/ACE_Medical_Actions.hpp]: You can use a macro for those long config classes.

Eg:
```
#define MACRO_BIOFOAM(var) \
    class Biofoam { \
        displayName = CSTRING(Biofoam); \
        distance = 2.0; \
        condition = QUOTE([ARR_4(_player, _target, var, 'Biofoam')] call DEFUNC(medical,canTreatCached)); \
        statement = QUOTE([ARR_4(_player, _target, var, 'Biofoam')] call DEFUNC(medical,treatment)); \
        exceptions[] = {}; \
        showDisabled = 1; \
        priority = 2; \
        hotkey = ""; \
        icon = PATHTOF(UI\icons\Biofoam.paa); \
    };

class ACE_Head {
    MACRO_BIOFOAM('head')
};
```

Or even with added body part:
```
#define MACRO_BIOFOAM(var1,var2) \
    class ACE_var1 { \
        class Biofoam { \
            displayName = CSTRING(Biofoam); \
            distance = 2.0; \
            condition = QUOTE([ARR_4(_player, _target, var2, 'Biofoam')] call DEFUNC(medical,canTreatCached)); \
            statement = QUOTE([ARR_4(_player, _target, var2, 'Biofoam')] call DEFUNC(medical,treatment)); \
            exceptions[] = {}; \
            showDisabled = 1; \
            priority = 2; \
            hotkey = ""; \
            icon = PATHTOF(UI\icons\Biofoam.paa); \
        }; \
    };

MACRO_BIOFOAM(Head,'head')
```

(Untested, could contain errors!)

File: optionals/compat_optre/cfgWeapons.hpp [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        descriptionUse = "Apply to affected area to stop bleeding and apply a all-natual pain killer.";
+        class ItemInfo: InventoryItem_Base_F {
+            mass = 8;
+        };
+    };
+    class EBR_Base_F;  // External class reference
+    class UGL_F;  // External class reference
+    class LMG_Mk200_F;  // External class reference
+	class Pistol_Base_F;  // External class reference
+	class Rifle_Base_F;  // External class reference
```
Comment (jonpas) [inline optionals/compat_optre/cfgWeapons.hpp]: Weird indentation.

File: optionals/compat_optre/functions/script_component.hpp [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+#ifdef DEBUG_ENABLED_COMPAT_OPTRE
+    #define DEBUG_MODE_FULL
+#endif
+
+#ifdef DEBUG_SETTINGS_COMPAT_OPTRE
+    #define DEBUG_SETTINGS DEBUG_SETTINGS_COMPAT_OPTRE
+#endif
+
+#include "\z\ace\addons\main\script_macros.hpp"
```
Comment (jonpas) [inline optionals/compat_optre/functions/script_component.hpp]: Replace entire file contents with: `#include "\z\ace\addons\compat_optre\script_component.hpp"`

**Comment by jonpas** [comment]
Lots of braces in new lines, they have to be in the same line:
```
class Bla {
    entry = value;
};
```

And **not**:
```
class Bla
{
    entry = value;
};
```

**Comment by Sgt-Parker** [comment]
I fixed the bracket issue for you.

File: optionals/compat_optre/functions/script_component.hpp (hunk 3 above)
Comment (Sgt-Parker) [inline optionals/compat_optre/functions/script_component.hpp]: What do you mean with this?

Comment (jonpas) [inline optionals/compat_optre/functions/script_component.hpp]: That this file only needs the line I wrote, no other definitions are needed.

Comment (Sgt-Parker) [inline optionals/compat_optre/functions/script_component.hpp]: Oh, I see.

**Comment by kripto202** [comment]
when entering the server with this compatibility pbo, I keep getting an ace error message stating server/client has extra addon ["ace_compat_optre"]. Can someone confirm?

**Comment by Glowbal** [comment]
Before this gets merged, the commits need to be cleaned up.

**Comment by Sgt-Parker** [comment]
I am new to the whole github thing, what do you mean by "clean up" Glowbal?

**Comment by MikeMatrix** [comment]
Here is a little to read up on "cleaning up" or in this case, squashing commits.
https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Squashing-Commits

**Comment by kripto202** [comment]
possibly add the srs99d to the list of gun profiles in the atragMX?

**Comment by kripto202** [comment]
Will need the interaction point, for the pelican, be moved from the top to somewhere close to the bottom or middle. The pelican is so tall that you can't reach the memory point.

**Comment by kripto202** [comment]
what is left to get this working?

**Comment by jonpas** [comment]
@Nathan423 This PR has a lot of similar things to #2859, reason?

**Comment by PabstMirror** [comment]
I believe the OPTRE devs are working on a native compat pbo

**Pull request closed without merging**